File: password_strength_Check.cpp
#include<iostream>
#include<regex>
#include<string>
#include<vector>

using namespace std;

pair<string, vector<string>>
passwordStrength(const string& password){
    int score = 0;
    vector<string> suggestions;


// check password length
    if(password.length()>=8){
        score++;
    }else{
        suggestions.push_back("Password should be at least 8 character long.");
    }

// check uppercase letter
    if(regex_search(password, regex("[A-Z]"))){
        score++;
    } else {
        suggestions.push_back("Password should contain at least one uppercase letter.");
    }

//check lowercase letter
    if(regex_search(password, regex("[a-z]"))){
        score++;
    } else {
        suggestions.push_back("Password should contain at least one lowercase letter.");
    }

//check digits
    if(regex_search(password, regex("[0-9]"))){
        score++;
    } else {
        suggestions.push_back("Password should be contain at least on edigit.");
    }

//check symbols
    if(regex_search(password, regex("[~!@#$%^&*(),.?\":{}|<>]"))) {
        score++;
    } else {
        suggestions.push_back("Password should be contain at least one special character.");
    }

// password strength determine
    string strength;
    if(score==5){
        strength = "Strong";
    } else if(score==4){
        strength = "Medium";
    } else {
        strength = "Weak";
    }

    return {strength, suggestions};
}


int main(){
    string password;
    cout<<"Enter your password:";
    cin>>password;

    auto[strength, suggestions]= passwordStrength(password);

    cout<<"Password Strength:"<<strength<<"\n";
    if(!suggestions.empty()){
        cout<<"\nSuggestions to improve your password:\n";
        for(const auto& suggestion:suggestions){
            cout<<"-"<<suggestion<<"\n";
        }
    } else {
        cout<<"Yore password is strong.\n";
    }
    return 0;
}

File: Password_strength_Check.py
import re

def password_strength(password):
    score = 0
    suggestions  =[]

    #check password length
    if len(password)>=8:
        score += 1
    else:
        suggestions.append("Password should be at least 8 characters long,")

    #check for uppercase letter
    if re.search(r'[A-Z]', password):
        score +=1
    else:
        suggestions.append("password should be contain atleast one uppercse letter.")

    #check for lowercase letter
    if re.search(r'[a-z]', password):
        score += 1
    else:
        suggestions.append("Password should be contain atleast one lowercase letter.")

    #check for digits
    if re.search(r'[0-9]', password):
        score +=1
    else:
        suggestions.append("Pasword should be contain atleast onr digit.")

    #check for special character
    if re.search(r'[!@#$%^&*(),.?:|<>]', password):
        score +=1
    else:
        suggestions.append("Password should contain at least one special character.")

    #Determine password strength
    if score == 5:
        strength = "Strong"
    elif score == 4:
        strength = "Medium"
    else:
        strength = "Weak"

    return strength, suggestions

# Test the function
password = input("Enter your password:")
strength, suggestions = password_strength(password)

print(f"Password Strength:{strength}")
if suggestions:
    print("\nSuggestions to improve your password:")
    for suggestion in suggestions:
        print(f"-{suggestion}")
else:
    print("Your password is strong.")